src/avram/migrator/create_index_statement.cr

```diff
@@ -28,10 +28,17 @@ require "./index_statement_helpers"
 class Avram::Migrator::CreateIndexStatement
   include Avram::Migrator::IndexStatementHelpers
 
-  ALLOWED_INDEX_TYPES = %w[btree]
+  enum IndexTypes
+    Btree
+    Hash
+    Gist
+    Gin
+    Brin
+  end
 
-  def initialize(@table : TableName, @columns : Columns, @using : Symbol = :btree, @unique = false, @name : String? | Symbol? = nil)
-    raise "index type '#{using}' not supported" unless ALLOWED_INDEX_TYPES.includes?(using.to_s)
+  def initialize(@table : TableName, @columns : Columns, using : Symbol = :btree, @unique = false, @name : String? | Symbol? = nil)
```

Comment:
I was going to make this use the enum instead of a Symbol, but due to how the Crystal autocast works with passing a Symbol, it would have been a breaking change. This can be an optimization for later.